tree/ntuple/v7/test/ntuple_show.cxx

```diff
@@ -333,45 +333,55 @@ TEST(RNTupleShow, Objects)
 
 TEST(RNTupleShow, Collections)
 {
+   using ROOT::Experimental::RRecordField;
+   using ROOT::Experimental::RVectorField;
+
    std::string rootFileName{"test_ntuple_show_collection.root"};
    std::string ntupleName{"Collections"};
    FileRaii fileGuard(rootFileName);
    {
+      struct MyStruct {
+         int myInt;
+         float myFloat;
+      };
+
       auto model = RNTupleModel::Create();
-      auto collection_model = RNTupleModel::Create();
-      auto int_field = collection_model->MakeField<int>("myInt");
-      auto float_field = collection_model->MakeField<float>("myFloat");
-      auto collection = model->MakeCollection("collection", std::move(collection_model));
-      auto ntuple = RNTupleWriter::Recreate(std::move(model), ntupleName, rootFileName);
-      *int_field = 0;
-      *float_field = 10.0;
-      collection->Fill();
-      *int_field = 1;
-      *float_field = 20.0;
-      collection->Fill();
-      ntuple->Fill();
+      std::vector<std::unique_ptr<RFieldBase>> leafFields;
+      leafFields.emplace_back(std::make_unique<RField<int>>("myInt"));
+      leafFields.emplace_back(std::make_unique<RField<float>>("myFloat"));
+      auto recordField = std::make_unique<RRecordField>("_0", leafFields);
+      auto collectionField = RVectorField::CreateUntyped("myCollection", std::move(recordField));
+      model->AddField(std::move(collectionField));
+      model->Freeze();
+
+      auto v = std::static_pointer_cast<std::vector<MyStruct>>(model->GetDefaultEntry().GetPtr<void>("myCollection"));
```

Comment:
This requires the alignment inside `MyStruct` and the one assigned within the collection field to be exactly the same.  Do we have a way to cross check?

Reply:
Yes, through `RRecordField::GetOffsets()`. I added a check in the test.